File: src/main/java/fi/finlit/edith/ui/services/svn/RevisionInfo.java
/*
 * Copyright (c) 2009 Mysema Ltd.
 * All rights reserved.
 *
 */
package fi.finlit.edith.ui.services.svn;

public class RevisionInfo {

    private final long svnRevision;

    private final String created;

    private final String creator;

    public RevisionInfo(long svnRevision, String created, String creator){
        this.svnRevision = svnRevision;
        this.created = created;
        this.creator = creator;
    }

    public RevisionInfo(long svnRevision){
        this.svnRevision = svnRevision;
        this.created = "";
        this.creator = "";
    }

    public long getSvnRevision() {
        return svnRevision;
    }

    public String getCreated() {
        return created;
    }

    public String getCreator() {
        return creator;
    }

    @Override
    public int hashCode() {
        return Long.valueOf(svnRevision).hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        RevisionInfo other = (RevisionInfo) obj;
        if (svnRevision != other.svnRevision) {
            return false;
        }
        return true;
    }
}

File: src/main/scala/fi/finlit/edith/ui/services/svn/RevisionInfo.scala
package fi.finlit.edith.ui.services.svn

import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

class RevisionInfo(@BeanProperty val svnRevision: Long, @BeanProperty val created: String, @BeanProperty val creator: String) {

  def this(svnRevision: Long) {
    this.svnRevision = svnRevision
    this.created = ""
    this.creator = ""
  }

  override def hashCode(): Int = Long.valueOf(svnRevision).hashCode

  override def equals(obj: Any): Boolean = {
    if (this == obj) {
      return true
    }
    if (obj == null) {
      return false
    }
    if (getClass != obj.getClass) {
      return false
    }
    var other = obj.asInstanceOf[RevisionInfo]
    if (svnRevision != other.svnRevision) {
      return false
    }
    true
  }
}
